packages/next/client/components/hooks-server.ts

```diff
@@ -3,16 +3,27 @@ import {
   HeadersContext,
   PreviewDataContext,
   CookiesContext,
+  DynamicServerError,
+  StaticGenerationContext,
 } from './hooks-server-context'
 
+function useStaticGenerationBailout() {
+  if (useContext(StaticGenerationContext)) {
+    throw new DynamicServerError('useHeader')
```

Comment:
Probably not the message you want to include here

Reply:
This isn't shown to the user currently more so meant for context if we want to show an explanation for static bail out in the future.

Reply:
Missed this got changed, fixed in https://github.com/vercel/next.js/pull/39884/commits/f66e75a3c87a319fa5ba570359c8fd4853d74301